pkg/yttlibrary/overlay/map.go

```diff
@@ -19,7 +21,7 @@ func (o OverlayOp) mergeMapItem(leftMap *yamlmeta.Map, newItem *yamlmeta.MapItem
 
 	leftIdx, found, err := ann.Index(leftMap)
 	if err != nil {
-		return err
+		return fmt.Errorf("Searching in map for item: %s", err)
```

Comment:
are changes to overlay library just general improvements? if so let's split that into a separate PR.

Reply:
we should also move err augmentation into Index().